app/src/common/shared/org/mozilla/vrbrowser/browser/SettingsStore.java

```diff
@@ -116,10 +117,17 @@ public void setTelemetryEnabled(boolean isEnabled) {
         final boolean hasEnabled = isTelemetryEnabled();
         if (hasEnabled != isEnabled) {
             TelemetryWrapper.init(mContext);
+            GleanMetricsService.init(mContext);
         }
 
         TelemetryHolder.get().getConfiguration().setUploadEnabled(isEnabled);
         TelemetryHolder.get().getConfiguration().setCollectionEnabled(isEnabled);
+
+        if (isEnabled) {
```

Comment:
This could just be `Glean.INSTANCE.setUploadEnabled(isEnabled);`

Reply:
I would still manage it behind the wrapper. I am also planning in unit tests for GleanService, I would think behinding the wrapper would be a good way to go.

Reply:
For your interest, the Glean SDK provides a [testing API](https://mozilla.github.io/glean/book/user/testing-metrics.html) for metrics and pings.